provider/azure-devops-workitems/index.ts

```diff
@@ -0,0 +1,92 @@
+import type {
+    Item,
+    ItemsParams,
+    ItemsResult,
+    MentionsParams,
+    MentionsResult,
+    MetaResult,
+    Provider,
+} from '@openctx/provider'
+import dedent from 'dedent'
+import { XMLBuilder } from 'fast-xml-parser'
+import { type SimpleWorkItem, fetchWorkItem, searchWorkItems } from './api.js'
+
+export type Settings = {
+    url: string
+    accessToken: string
+}
+
+const xmlBuilder = new XMLBuilder({ format: true })
+
+const checkSettings = (settings: Settings) => {
+    const missingKeys = ['url', 'accessToken'].filter(key => !(key in settings))
+    if (missingKeys.length > 0) {
+        throw new Error(`Missing settings: ${JSON.stringify(missingKeys)}`)
+    }
+}
+
+const wiToItem = (workItem: SimpleWorkItem): Item => ({
+    url: workItem.url,
+    title: workItem.fields.title,
+    ui: {
+        hover: {
+            markdown: workItem.fields.description,
+            text: workItem.fields.description || workItem.fields.title,
+        },
+    },
+    ai: {
+        content: dedent`
+            The following represents contents of the Azure DevOps Work Item ${workItem.id}.  
+            Use it to check if it helps.
+            Ignore it if it is not relevant.
+
+            ${xmlBuilder.build({
+                id: workItem.id,
+                state: workItem.fields.state,
+                assignedTo: workItem.fields.assignedTo,
+                type: workItem.fields.type,
+                url: workItem.url,
+                description: workItem.fields.description,
+                tags: workItem.fields.tags,
+            })}
+            `,
+    },
+})
+
+const azureDevOps: Provider = {
+    meta(): MetaResult {
+        return { name: 'Azure DevOps Work Items', mentions: {} }
+    },
+
+    async mentions(params: MentionsParams, settings: Settings): Promise<MentionsResult> {
+        checkSettings(settings)
+
+        // Uses the quick REST picker API to fuzzy match potential items
+        return searchWorkItems(params.query, settings).then(items =>
+            items.map(item => ({
+                title: item.title,
+                uri: item.url,
+                description: item.description,
+                data: { id: item.id },
```

Comment:
any chance `item.url` contains the ID? May be a more robust way to do this, since a future feature we will have is being able to paste in urls and directly get the items.

Reply:
It does, but in that case you mean we would pass the url from here and into the items method as the identifier? And then parse our the id?

Reply:
I think the URL pattern matching support would use some other type of meta regex registration @keegancsmith? I'm not sure we need to worry about the data. fwiw I pass only the ID through data with the Jira and Confluence providers too.